packages/overlay/src/integrations/sentry/components/insights/FunctionProfiles.tsx

```diff
@@ -0,0 +1,111 @@
+import { useMemo } from 'react';
+import { ReactComponent as Sort } from '~/assets/sort.svg';
+import { ReactComponent as SortDown } from '~/assets/sortDown.svg';
+import classNames from '~/lib/classNames';
+import { FUNCTION_PROFILES_HEADERS, FUNCTION_PROFILES_SORT_KEYS } from '../../constants';
+import useSentryStore from '../../data/sentryStore';
+import useSort from '../../hooks/useSort';
+import type { FunctionProfile } from '../../types';
+import { getFormattedDuration, getSpanDurationClassName } from '../../utils/duration';
+import { TimeBar } from '../shared/TimeBar';
+
+type FunctionProfileComparator = (a: FunctionProfile, b: FunctionProfile) => number;
+type FunctionProfileSortTypes = (typeof FUNCTION_PROFILES_SORT_KEYS)[keyof typeof FUNCTION_PROFILES_SORT_KEYS];
+
+const COMPARATORS: Record<FunctionProfileSortTypes, FunctionProfileComparator> = {
+  [FUNCTION_PROFILES_SORT_KEYS.functionName]: (a, b) => {
+    if (a.name < b.name) return -1;
+    if (a.name > b.name) return 1;
+    return 0;
+  },
+  [FUNCTION_PROFILES_SORT_KEYS.timeSpent]: (a, b) => a.totalTime - b.totalTime,
+  [FUNCTION_PROFILES_SORT_KEYS.samples]: (a, b) => a.samples - b.samples,
+  [FUNCTION_PROFILES_SORT_KEYS.profiles]: (a, b) => {
+    if (a.traceId < b.traceId) return -1;
+    if (a.traceId > b.traceId) return 1;
+    return 0;
+  },
+};
+
+function FunctionProfiles() {
+  const { sort, toggleSortOrder } = useSort({ defaultSortType: FUNCTION_PROFILES_SORT_KEYS.timeSpent });
+
+  const functionProfiles = useMemo(() => {
+    const profiles = useSentryStore.getState().getFunctionProfiles();
+    const compareProfileInfo = COMPARATORS[sort.active] || COMPARATORS[FUNCTION_PROFILES_SORT_KEYS.timeSpent];
+
+    return profiles.sort((a, b) => {
+      return sort.asc ? compareProfileInfo(a, b) : compareProfileInfo(b, a);
+    });
+  }, [sort]);
+
+  if (!functionProfiles?.length) {
+    return <p className="text-primary-300 px-6 py-4">No profiles found.</p>;
+  }
+
+  // Calculate max time for bar visualization (100%, scaling form here)
+  const maxTime = Math.max(...functionProfiles.map(profile => profile.totalTime));
+
+  return (
+    <table className="divide-primary-700 w-full table-fixed divide-y">
```

Comment:
We'll need to switch to `<Table>` from `import Table from '~/ui/Table';` for this.